server/src/main/java/swim/tutorial/UnitAgent.java

```diff
@@ -12,14 +12,115 @@
 import java.util.Iterator;
 
 public class UnitAgent extends AbstractAgent {
-
-  @SwimLane("histogram")
-  private final MapLane<Long, Value> histogram = this.<Long, Value>mapLane()
-      .didUpdate((k, n, o) -> {
-        logMessage("histogram: replaced " + k + "'s value to " + Recon.toString(n) + " from " + Recon.toString(o));
-        dropOldData();
-      });
-
+	
+	  // *********************** EXAMPLE SOLUTIONS FOR STATS LANE ***********************
+	  
+	  // instance variables to track metrics going into stats
+	  private long countSum = 0;
+	  private int countTotal = 0;
+	  private int index = 0;
+	  private long[] recentData = new long[5];
+	  
+	  
+	  // intermediary lanes that represent individual metrics
+	  @SwimLane("avg")
+	  private final ValueLane<Long> avg = this.<Long>valueLane()
+	  .didSet((n, o) -> {
+	    logMessage("avg: mean updated to " + n + " from " + o);
+	  });
+	  
+	  @SwimLane("localAvg")
+	  private final ValueLane<Long> localAvg = this.<Long>valueLane()
+	  .didSet((n, o) -> {
+	    logMessage("localAvg: local average (last 5 entries) updated to " + n + " from " + o);
+	  });
+	  
+	  @SwimLane("localVar")
+	  private final ValueLane<Long> localVar = this.<Long>valueLane()
+	  .didSet((n, o) -> {
+	    logMessage("localVar: local variance (last 5 entries) updated to " + n + " from " + o);
+	  });
+	  
+	  @SwimLane("localStdDev")
+	  private final ValueLane<Long> localStdDev = this.<Long>valueLane()
+	  .didSet((n, o) -> {
+	    logMessage("localStdDev: local std deviation (last 5 entries) updated to " + n + " from " + o);
+	  });
+	  
+	  
+	  // combination all calculations into one swim lane of type Value
+	  @SwimLane("stats")
+	  private final ValueLane<Value> stats = this.<Value>valueLane()
+	  .didSet((n, o) -> {
+		  logMessage("stats: set to " + Recon.toString(n) + " from " + Recon.toString(o));
+	  });
+	
+	  // *********************** EXAMPLE SOLUTION FOR HISTOGRAM ***********************
+	  @SwimLane("histogram")
+	  private final MapLane<Long, Value> histogram = this.<Long, Value>mapLane()
+	      .didUpdate((k, n, o) -> {
+	        logMessage("histogram: replaced " + k + "'s value to " + Recon.toString(n) + " from " + Recon.toString(o));
+	        
+	        // calculating overall mean to send to average lane
+	        countSum += n.get("count").longValue();
+	        countTotal ++;
+	        final long setAvg = countSum / countTotal;
+	        avg.set(setAvg);
+	        
+	        // appending new data to the recentData array
+	        if (index >= recentData.length-1) {
+	        	index = 0;
+	        }
+	        recentData[index] = n.get("count").longValue();
+	        index ++;
```

Comment:
@ajay-gov thoughts on whether this should be `index++` (no space) instead? I _think_ we had a `checkstyle.xml` in another project that encouraged this pattern?